packages/settingeditor/src/settingspanel.tsx

```diff
@@ -79,6 +94,25 @@ export const SettingsPanel: React.FC<ISettingsPanelProps> = ({
     [id: string]: boolean;
   }> = React.useRef({});
 
+  // When filter updates, only show plugins that match search.
+  updateFilterSignal.connect(
```

Comment:
You need to wrap this in a `useEffect` (you can reuse the one below).

The current code will be called at every react update cycle. So it will create multiple connections without cleaning them. Use `useEffect` with empty array as second argument (as the one L116) will connect the signal only when mounting the component.

> The `useEffect` should return a callback that is disconnecting the signal. That callback is triggered when unmounting the component (as the second argument of `useEffect` is an empty array).